src/freon.c

```diff
@@ -218,9 +218,11 @@ static int _open(int (*origfunc)(const char *pathname, int flags, mode_t mode),
 int open(const char *pathname, int flags, ...) {
     if (!orig_open) preload_init();
 
+    mode_t mode = 0;
     va_list argp;
     va_start(argp, flags);
-    mode_t mode = va_arg(argp, mode_t);
+    if (flags & (O_CREAT | O_TMPFILE))
```

Comment:
O_TMPFILE apparently isn't defined in precise.  I think you need some sort of compatibility define.